geoapps/base/application.py

```diff
@@ -305,6 +278,7 @@ def get_output_workspace(self, workpath: str = "./", name: str = "Temp.geoh5"):
             name += ".geoh5"
 
         workspace = Workspace(path.join(workpath, name))
+        workspace.close()
```

Comment:
Closing the file wipes the class of entities but keeps track of the h5file. As soon as you try to do an operation again on it, it will re-build itself in mode "r+".